src/audio-worklet/basic/noise-generator/main.js

```diff
@@ -20,15 +22,29 @@ const startAudio = async (context) => {
   modulator.start();
 };
 
-// A simplem onLoad handler. It also handles user gesture to unlock the audio
+const stopAudio = () => {
+  if (noiseGenerator) {
+    noiseGenerator.disconnect();
+    isPlaying = false;
+  }
+};
+
+// A simple onLoad handler. It also handles user gesture to unlock the audio
 // playback.
 window.addEventListener('load', async () => {
   const buttonEl = document.getElementById('button-start');
   buttonEl.disabled = false;
+
   buttonEl.addEventListener('click', async () => {
-    await startAudio(audioContext);
-    audioContext.resume();
-    buttonEl.disabled = true;
-    buttonEl.textContent = 'Playing...';
-  }, false);
+    if (!isPlaying) { 
+      // If not playing, start the audio.
+      await startAudio(audioContext);
```

Comment:
This will call L10 (addModule) multiple times depending on `isPlaying`. Have you tested locally? Loading the same script twice should throw an exception.

Reply:
I have updated the code to ensure that the module is added only once by using a flag (moduleAdded) to track whether the module has already been loaded.

Reply:
I believe this example is pretty much the same with the other oscillator example. Why do we need a different flag (`moduleAdded`) only for this example?